### PR TITLE
Save the scope of an auth token in the session

### PR DESCRIPTION
**Pull request by icewind1991**
And re-use the saved scope when no other scope has been set

### CODE REVIEW

**Review by LukasReschke — CHANGES_REQUESTED** [review]
I'll add an integration test to that before we merge it.

File: lib/private/Lockdown/LockdownManager.php [hunk 1]
```diff
@@ -20,27 +20,60 @@
...
+		$callback = $this->sessionCallback;
+		return $callback();
+	}
+
+	private function getScopeAsArray() {
+		if (!$this->scope) {
+			$session = $this->getSession();
+			$sessionScope = $session->get('token_scope');
+			if ($sessionScope) {
+				$this->scope = $sessionScope;
```
Comment (LukasReschke) [inline lib/private/Lockdown/LockdownManager.php]: Not covered by tests :)

**Review by LukasReschke — APPROVED** [review]
Added integration tests for this with https://github.com/nextcloud/server/pull/3526/commits/1d3e391ad8b9eadefab8eaa8239e6901b8466f01. Let's see if they pass :)

**Review by rullzer — APPROVED** [review]

**Comment by LukasReschke** [comment]
stable11: https://github.com/nextcloud/server/pull/4225

**Pull request merged**